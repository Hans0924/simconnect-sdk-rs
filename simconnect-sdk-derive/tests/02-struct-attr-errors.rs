#![allow(unused_variables, dead_code)]
use simconnect_sdk_derive::SimConnectObject;

#[derive(Debug, Clone, SimConnectObject)]
#[simconnect(period = "visual-frame", condition = "changed")]
struct GpsData1(f64);

#[derive(Debug, Clone, SimConnectObject)]
struct GpsData2 {}

#[derive(Debug, Clone, SimConnectObject)]
#[simconnect]
struct GpsData3 {}

#[derive(Debug, Clone, SimConnectObject)]
#[simconnect()]
struct GpsData4 {}

#[derive(Debug, Clone, SimConnectObject)]
#[simconnect(period = "second", period = "second")]
struct GpsData5 {}

#[derive(Debug, Clone, SimConnectObject)]
#[simconnect(condition = "none", condition = "none")]
struct GpsData6 {}

#[derive(Debug, Clone, SimConnectObject)]
#[simconnect(period = "second", condition = "none", test = "test")]
struct GpsData7 {}

#[derive(Debug, Clone, SimConnectObject)]
#[simconnect(periodX = "second", condition = "none")]
struct GpsData8 {}

#[derive(Debug, Clone, SimConnectObject)]
#[simconnect(period = "second", conditionX = "none")]
struct GpsData9 {}

fn main() {}
