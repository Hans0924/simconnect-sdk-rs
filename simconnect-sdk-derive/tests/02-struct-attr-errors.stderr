error: Unsupported field type. Only named fields are supported.
 --> tests/02-struct-attr-errors.rs:5:1
  |
5 | / #[simconnect(period = "visual-frame", condition = "changed")]
6 | | struct GpsData1(f64);
  | |_____________________^

error: expected attribute `#[simconnect(period = "...", condition = "...")]`
 --> tests/02-struct-attr-errors.rs:9:1
  |
9 | struct GpsData2 {}
  | ^^^^^^^^^^^^^^^^^^

error: expected attribute `#[simconnect(period = "...", condition = "...")]`
  --> tests/02-struct-attr-errors.rs:12:3
   |
12 | #[simconnect]
   |   ^^^^^^^^^^

error: expected attribute `#[simconnect(period = "...", condition = "...")]`
  --> tests/02-struct-attr-errors.rs:16:3
   |
16 | #[simconnect()]
   |   ^^^^^^^^^^^^

error: expected attribute `#[simconnect(period = "...", condition = "...")]`
  --> tests/02-struct-attr-errors.rs:20:3
   |
20 | #[simconnect(period = "second", period = "second")]
   |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: expected attribute `#[simconnect(period = "...", condition = "...")]`
  --> tests/02-struct-attr-errors.rs:24:3
   |
24 | #[simconnect(condition = "none", condition = "none")]
   |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: expected attribute `#[simconnect(period = "...", condition = "...")]`
  --> tests/02-struct-attr-errors.rs:28:3
   |
28 | #[simconnect(period = "second", condition = "none", test = "test")]
   |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: expected attribute `#[simconnect(period = "...", condition = "...")]`
  --> tests/02-struct-attr-errors.rs:32:3
   |
32 | #[simconnect(periodX = "second", condition = "none")]
   |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: expected attribute `#[simconnect(period = "...", condition = "...")]`
  --> tests/02-struct-attr-errors.rs:36:3
   |
36 | #[simconnect(period = "second", conditionX = "none")]
   |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
